### ISSUE TITLE
Type Datetime annotation timezone

### ISSUE THREAD

**Issue by jean553** [schmittjoh/JMSSerializerBundle#728]
Hi!

I have an entity that stores a datetime, and I use the JMSSerializer in order to return this field through an API:

```php
/**
 * @ORM\Column(type="datetime")
 *
 * @Serializer\Type("DateTime<'d/m - H:i', 'Europe/Paris'>")
 */
private $creationDateTime;
```

Into the database, the creationDateTime is stored with the UTC timezone.

Changing the format is correctly applied to the JSON API response (for instance, replacing "d/m H:i by d/m/Y H:i"), but the returned time is always the stored one (UTC one) and not the expected one (Europe/Paris) even if I set a totally different timezone.

For example, if the stored time is 15:04, the serialized time returned by the API is still 15:04 (UTC) and not the expected one (16:04, Europe/Paris).

Is there any way to apply the timezone on the stored date when serializing the response ?

I use: Symfony 4, JMS-SerializerBundle 3.1.0, JMS-Serializer 2.1.0

Here the walk-around I made, but definitely not the best way to solve this issue:

```php
/**
 * Returns the signal creation date and time
 *
 * @Serializer\Type(
 *     "DateTime<'d/m/Y H:i'>"
 * )
 *
 * @Serializer\Groups({"signals"})
 * @Serializer\VirtualProperty()
 * @Serializer\Expose
 */
public function getCreationDateTime(): \DateTime
{
    $dateTime = $this->creationDateTime;
    $dateTime->setTimeZone(new \DateTimeZone("Europe/Paris"));

    return $dateTime;
}
```

Thanks in advance.

**Comment by goetas** [comment]
Any reason why the expected time zone is Paris?

If  your application uses Paris as default time zone, shouldn't be stored in dB as Paris? 

If your app stores dates effectively in utc, then converting it to paris is an application requirement, thus should stay in application code 

Did i miss something?

**Comment by realjjaveweb** [comment]
@goetas yes you did, the UTC offset only tells you the technical timezone, but, it does not say anything about the local timezone (as timezone string does), which becomes a problem when that local timezone has a DST (daylight saving time) (or really any other local-specific time shift for the future).

There are 2 main solutions to this:
1. custom datetime format that includes this - which in [PHP's datetime format](https://www.php.net/manual/en/datetime.format.php) is small letter "`e`" - but this is very custom and other languages may have trouble parsing this out (just assuming here)
2. separate field(or even column) for timezone(string) (which is something I saw in practice a few times and always successful)

**Comment by realjjaveweb** [comment]
However so far I failed to find how to describe DateTimeZone => string and string => DateTimeZone for JMS serializer in PHP ...
